test/integration/conftest.py

```diff
@@ -199,35 +203,16 @@ def _create_autosubmit_exp(
             expid=expid,
             basic_config=BasicConfig
         )
+        config.reload(force_load=True)
 
-        config.experiment_data = {**config.experiment_data, **experiment_data}
-
-        key_file = {
-            'JOBS': 'jobs',
-            'PLATFORMS': 'platforms',
-            'EXPERIMENT': 'expdef'
-        }
-
-        for key, input_lines in key_file.items():
-            if key in experiment_data:
-                mode = 'a' if key == 'EXPERIMENT' else 'w'
-                with open(conf_dir / f'{input_lines}_{expid}.yml', mode) as f:
-                    YAML().dump({key: experiment_data[key]}, f)
-
-        other_yaml = {
-            k: v for k, v in experiment_data.items()
-            if k not in key_file
-        }
-        if other_yaml:
-            with open(conf_dir / f'tests_{expid}.yml', 'w') as f:
-                YAML().dump(other_yaml, f)
-
-        if reload:
-            config.reload(force_load=True)
+        # Remove original files. So we can save a new file with all the memory modifications.
+        for f in conf_dir.iterdir():
+            if f.is_file():
+                f.unlink()
```

Comment:
This is more error-prone, I think. I mean, re-using the directory. Each test should start from scratch if possible. In which occasion did you have the same experiment re-used in an integration test?